test/api/api_test.go

```diff
@@ -138,13 +140,77 @@ func TestWithDBSession(t *testing.T) {
 	}
 }
 
+// TestWithDBSession_RetryOnWriteConflict ensures that WithDBSession can
+// properly retry requests on MongoDB WriteConflict error.
+func TestWithDBSession_RetryOnWriteConflict(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	dbName := test.DBNameForTest(t.Name())
+	dep := lib.NewDependencyUserPutMongoDelay()
```

Comment:
It's always good to drop a comment above the dependency you inject that indicates what it does and why you need it. It's not clear to me why you need to delay the write for 100ms in 50% of the cases for this test to pass.

Reply:
Added comment, removed randomness.